cf-oscore/src/main/java/org/eclipse/californium/oscore/OptionJuggle.java

```diff
@@ -328,4 +329,73 @@ public static OptionSet merge(OptionSet eOptions, OptionSet uOptions) {
 		return eOptions;
 	}
 
+	/**
+	 * Retrieve RID value from an OSCORE option.
+	 * 
+	 * @param oscoreOption the OSCORE option
+	 * @return the RID value
+	 */
+	static byte[] getRid(byte[] oscoreOption) {
+		if (oscoreOption.length == 0) {
+			oscoreOption = new byte[] { 0x00 };
+		}
+	
+		// Parse the flag byte
+		byte flagByte = oscoreOption[0];
+		int n = flagByte & 0x07;
+		int k = flagByte & 0x08;
+		int h = flagByte & 0x10;
+	
+		byte[] kid = null;
+		int index = 1;
+	
+		// Partial IV
+		index += n;
+	
+		// KID Context
+		if (h != 0) {
+			int s = oscoreOption[index];
+			index += s + 1;
+		}
+	
+		// KID
+		if (k != 0) {
+			kid = Arrays.copyOfRange(oscoreOption, index, oscoreOption.length);
+		}
+	
+		return kid;
+	}
+
+	/**
+	 * Retrieve ID Context value from an OSCORE option.
+	 * 
+	 * @param oscoreOption the OSCORE option
+	 * @return the ID Context value
+	 */
+	static byte[] getIDContext(byte[] oscoreOption) {
+		if (oscoreOption.length == 0) {
+			oscoreOption = new byte[] { 0x00 };
```

Comment:
For me this case will always return `null`, or?

Reply:
Yes, you mean that I could return null straight in that if, so having?
```
static byte[] getIDContext(byte[] oscoreOption) {
		if (oscoreOption.length == 0) {
			return null;
}
```
(And same for getRID).

Reply:
I updated to return null immediately in commit 9a654a3f67270da32a728120f62cfc0e5919d242